Mathlib/CategoryTheory/Bicategory/Adjunction/Mate.lean

```diff
@@ -65,58 +135,27 @@ Then we have a bijection between natural transformations `g ≫ l₂ ⟶ l₁ 
-  left_inv α :=
-    calc
-      _ = 𝟙 _ ⊗≫ (adj₁.unit ▷ (g ≫ 𝟙 e) ≫ (l₁ ≫ r₁) ◁ g ◁ adj₂.unit) ▷ l₂ ⊗≫
-            l₁ ◁ r₁ ◁ α ▷ r₂ ▷ l₂ ⊗≫
-              l₁ ◁ (adj₁.counit ▷ h ▷ (r₂ ≫ l₂) ≫ (𝟙 d ≫ h) ◁ adj₂.counit) ⊗≫ 𝟙 _ := by
-        bicategory
-      _ = 𝟙 _ ⊗≫ g ◁ adj₂.unit ▷ l₂ ⊗≫
-            (adj₁.unit ▷ (g ≫ l₂) ≫ (l₁ ≫ r₁) ◁ α) ▷ (r₂ ≫ l₂) ⊗≫
-              l₁ ◁ (((r₁ ≫ l₁) ≫ h) ◁ adj₂.counit ≫ adj₁.counit ▷ h ▷ 𝟙 f) ⊗≫ 𝟙 _ := by
-        rw [← whisker_exchange, ← whisker_exchange]
-        bicategory
-      _ = 𝟙 _ ⊗≫ g ◁ adj₂.unit ▷ l₂ ⊗≫ α ▷ r₂ ▷ l₂ ⊗≫
-            leftZigzag adj₁.unit adj₁.counit ▷ h ▷ r₂ ▷ l₂ ⊗≫ l₁ ◁ h ◁ adj₂.counit ⊗≫ 𝟙 _ := by
-        rw [← whisker_exchange, whisker_exchange _ adj₂.counit]
-        bicategory
-      _ = 𝟙 _ ⊗≫ g ◁ adj₂.unit ▷ l₂ ⊗≫ (α ▷ (r₂ ≫ l₂) ≫ (l₁ ≫ h) ◁ adj₂.counit) ⊗≫ 𝟙 _ := by
-        rw [adj₁.left_triangle]
-        bicategory
-      _ = 𝟙 _ ⊗≫ g ◁ (leftZigzag adj₂.unit adj₂.counit) ⊗≫ α ⊗≫ 𝟙 _ := by
-        rw [← whisker_exchange]
-        bicategory
-      _ = α := by
-        rw [adj₂.left_triangle]
-        bicategory
-  right_inv β :=
-    calc
-      _ = 𝟙 _ ⊗≫ r₁ ◁ ((𝟙 c ≫ g) ◁ adj₂.unit ≫ adj₁.unit ▷ g ▷ (l₂ ≫ r₂)) ⊗≫
-            r₁ ◁ l₁ ◁ β ▷ l₂ ▷ r₂ ⊗≫
-              ((r₁ ≫ l₁) ◁ h ◁ adj₂.counit ≫ adj₁.counit ▷ (h ≫ 𝟙 f)) ▷ r₂ ⊗≫ 𝟙 _ := by
-        bicategory
-      _ = 𝟙 _ ⊗≫ r₁ ◁ (adj₁.unit ▷ g ▷ 𝟙 e ≫ ((l₁ ≫ r₁) ≫ g) ◁ adj₂.unit) ⊗≫
-            ((r₁ ≫ l₁) ◁ β ≫ adj₁.counit ▷ (h ≫ r₂)) ▷ l₂ ▷ r₂ ⊗≫
-              h ◁ adj₂.counit ▷ r₂ ⊗≫ 𝟙 _ := by
-        rw [whisker_exchange, whisker_exchange]
-        bicategory
-      _ = 𝟙 _ ⊗≫ r₁ ◁ g ◁ adj₂.unit ⊗≫ rightZigzag adj₁.unit adj₁.counit ▷ g ▷ l₂ ▷ r₂ ⊗≫
-            β ▷ l₂ ▷ r₂ ⊗≫ h ◁ adj₂.counit ▷ r₂ ⊗≫ 𝟙 _ := by
-        rw [whisker_exchange, ← whisker_exchange _ adj₂.unit]
-        bicategory
-      _ = 𝟙 _ ⊗≫ ((r₁ ≫ g) ◁ adj₂.unit ≫ β ▷ (l₂ ≫ r₂)) ⊗≫ h ◁ adj₂.counit ▷ r₂ ⊗≫ 𝟙 _ := by
-        rw [adj₁.right_triangle]
-        bicategory
-      _ = 𝟙 _ ⊗≫ β ⊗≫ h ◁ rightZigzag adj₂.unit adj₂.counit ⊗≫ 𝟙 _ := by
-        rw [whisker_exchange]
-        bicategory
-      _ = β := by
-        rw [adj₂.right_triangle]
-        bicategory
+@[simps! -isSimp]
+def mateEquiv : (g ≫ l₂ ⟶ l₁ ≫ h) ≃ (r₁ ≫ g ⟶ h ≫ r₂) :=
+  adj₂.homEquiv₂.trans ((Iso.homCongr (Iso.refl _) (α_ _ _ _)).trans adj₁.homEquiv₁)
+
+lemma mateEquiv_eq_iff (α : g ≫ l₂ ⟶ l₁ ≫ h) (β : r₁ ≫ g ⟶ h ≫ r₂) :
+    mateEquiv adj₁ adj₂ α = β ↔
+    adj₁.homEquiv₁.symm β = adj₂.homEquiv₂ α ≫ (α_ _ _ _).hom := by
+  conv_lhs => rw [eq_comm, ← adj₁.homEquiv₁.symm.injective.eq_iff']
+  rw [mateEquiv_apply, Equiv.symm_apply_apply]
+
+lemma mateEquiv_apply' (α : g ≫ l₂ ⟶ l₁ ≫ h) :
+    mateEquiv adj₁ adj₂ α =
+    r₁ ◁ (𝟙 g ≫ ((ρ_ g).inv ≫ g ◁ adj₂.unit ≫ (α_ g l₂ r₂).inv ≫ α ▷ r₂) ≫ (α_ l₁ h r₂).hom) ≫
+    (α_ r₁ l₁ (h ≫ r₂)).inv ≫ adj₁.counit ▷ (h ≫ r₂) ≫ (λ_ (h ≫ r₂)).hom :=
+  rfl
```

Comment:
Actually I'd prefer adding the following (non-defeq) lemma:
```lean
mateEquiv adj₁ adj₂ α = 𝟙 _ ⊗≫ r₁ ◁ g ◁ adj₂.unit ⊗≫ r₁ ◁ α ▷ r₂ ⊗≫ adj₁.counit ▷ h ▷ r₂ ⊗≫ 𝟙 _
```
where the expression is written so that we don't care about how the parentheses are placed or how the associators are inserted.

Reply:
Actually, terms with `⊗≫` create issues in downstream applications because it may create timeouts. For example, in a branch dealing with pseudofunctors to the bicategory of adjunctions, I had to replace `compUnit` (from `Adjunction.Basic`) so that it contains explicit associators, etc. In the proofs of the API for general abstract bicategories, it does not matter how we write the RHS because `bicategory` would work anyway.

Reply:
Ok, that looks good!
By the way, I'd like to look into timeout from `⊗≫`. Where can I find the branch for downstream applications?

Reply:
It seems I am no longer able to reproduce the timeouts... I have changed the lemmas so that they use `⊗≫` as you suggest, and I will ping you if timeouts issues reappear...